### PR TITLE
Replace hipblas API with rocBLAS

### PR DESCRIPTION
**Pull request by iotamudelta**
Requires some minor code changes to support rocBLAS's actual strided_batched API (the batched gemm API in HIP was only a loop over individual rocblas_gemm calls....). Still no fp16 support.

@petrex and @dagamayank for awareness

### CODE REVIEW

**Comment by iotamudelta** [comment]
@jihunnair-amd mind to review this for a second pair of eyes? thanks!

**Comment by iotamudelta** [comment]
@mwootton this means we can drop the hipblas install from the docker file you are working on

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -537,7 +537,7 @@ if(BUILD_CAFFE2 OR BUILD_ATEN)
     hip_include_directories(${Caffe2_HIP_INCLUDES})
 
     set(Caffe2_HIP_DEPENDENCY_LIBS
-      ${rocrand_LIBRARIES} ${hiprand_LIBRARIES} ${PYTORCH_HIP_HCC_LIBRARIES} ${PYTORCH_MIOPEN_LIBRARIES} ${hipblas_LIBRARIES})
+      ${rocrand_LIBRARIES} ${hiprand_LIBRARIES} ${PYTORCH_HIP_HCC_LIBRARIES} ${PYTORCH_MIOPEN_LIBRARIES})
```
Comment (jithunnair-amd) [inline cmake/Dependencies.cmake]: We don't need a `${rocblas_LIBRARIES}` here?

Comment (jithunnair-amd) [reply cmake/Dependencies.cmake]: Rest looks good afaict

Comment (iotamudelta) [reply cmake/Dependencies.cmake]: No, we don't, it adds rocBLAS a few lines below anyways (hipBLAS always relied on rocBLAS in the backend).

**Pull request merged**